src/ServiceProviders/RepositoryProviders/GitHub/GitHubProvider.php

```diff
@@ -255,12 +255,106 @@ protected function gitHubAPI($uri, $data = [], $method = 'GET')
         $message = isset($resultData['message']) ? "{$resultData['message']}." : '';
 
         if (!empty($message) || !empty($errors)) {
-            throw new TerminusException('{service} error: {message} {errors}', ['service' => $service, 'message' => $message, 'errors' => implode("\n", $errors)]);
+            throw new TerminusException('error: {message} {errors}', ['message' => $message, 'errors' => implode("\n", $errors)]);
         }
 
+        // The request may be against a paged collection. If that is the case, traverse the "next" links sequentially
+        // (since it's simpler and PHP doesn't have non-blocking I/O) until the end and accumulate the results.
+        $headers = $res->getHeaders();
+        // Check if the array is numeric. Otherwise we can't consider this a collection. Ideally GitHub would tell us
+        // that the response is a collection but we'll need to guess from the response format.
+        if ($follow_next_link && $this->isSequentialArray($resultData) && $this->isPagedResponse($headers)) {
+            $pager_info = $this->getPagerInfo($headers['Link']);
+            if (!$this->isLastPage($uri, $pager_info)) {
+                $next_page_uri = $this->getNextPageUri($pager_info);
+                var_dump($next_page_uri);
+                // Request the next page and append the data.
+                $resultData = array_merge_recursive(
+                    $resultData,
+                    $this->gitHubAPI($next_page_uri, $data, $method, TRUE)
+                );
+            }
+        }
         return $resultData;
     }
 
+    protected function isSequentialArray($input)
```

Comment:
The way we parse JSON responses will always yield a PHP array. This method checks if the array comes from `{foo: 'bar'}` or from `['foo', 'bar']`. Only the latter will be considered a response to a _collection endpoint that can be paginated_.